src/rpcmining.cpp

```diff
@@ -26,6 +26,56 @@
 
 using namespace std;
 
+bool CheckIsMiningSetup()
+{
+    if (!Params().MiningRequiresConfiguration())
+        return true;
+
+    if (!mapArgs.count("-blockminsize"))
+        return false;
+    if (!mapArgs.count("-blockmaxsize"))
+        return false;
+    if (!mapArgs.count("-blockprioritysize"))
+        return false;
+
+    // Either datacarriersize must be set, or datacarrier must be disabled
+    if (!(mapArgs.count("-datacarriersize") || !GetBoolArg("-datacarrier", true)))
+        return false;
+
+    return true;
+}
+
+void CheckMiningSetup(const char * const strWhatAmI)
+{
+    static bool fIsMiningSetup = false;
+    if (!fIsMiningSetup)
+        fIsMiningSetup = CheckIsMiningSetup();
+    if (fIsMiningSetup)
+        return;
+
+    unsigned long minsz = insecure_rand() % MAX_BLOCK_SIZE;
+    unsigned long maxsz = std::max((unsigned long)250000, minsz);
+    maxsz = maxsz + (insecure_rand() % (MAX_BLOCK_SIZE - maxsz));
+    unsigned long prisz = insecure_rand() % MAX_BLOCK_SIZE;
+    unsigned long datacarrier = insecure_rand() % 2;
+    unsigned long datacarriersize = insecure_rand() % 0x100;
+    throw JSONRPCError(RPC_NOT_CONFIGURED, strprintf(
```

Comment:
Can't you just use the defaults here and show a warning when the miner hasn't configured certain options instead for now? @jgarzik would that be acceptable?